src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs

```diff
@@ -24,12 +24,46 @@ internal DataOperationsCatalog(IHostEnvironment env)
             Environment = env;
         }
 
+        /// <summary>
+        /// Take an approximate bootstrap sample of <paramref name="input"/>.
+        /// </summary>
+        /// <remarks>
+        /// This sampler is a streaming version of a bootstrap sampler. Instead of taking the whole dataset into memory and resampling,
+        /// <see cref="BootstrapSample"/> streams through the dataset and uses a Poisson(1) distribution to select rows to add to the sample.
+        /// The <paramref name="complement"/> parameter allows for the creation of a bootstap sample and complementary out-of-bag sample by
+        /// using the same <paramref name="seed"/>.
+        /// </remarks>
+        /// <param name="input">The input data.</param>
+        /// <param name="complement">Whether this is the out-of-bag sample, that is, all those rows that are not selected by the transform.
+        /// Can be used to create a complementary pair of samples by using the same seed.</param>
+        /// <param name="seed">The random seed. If unspecified random state will be instead derived from the environment.</param>
+        /// <param name="shuffleInput">Whether we should attempt to shuffle the source data. By default on, but can be turned off for efficiency.</param>
+        /// <param name="poolSize">When shuffling the output, the number of output rows to keep in that pool. Note that shuffling of output is completely distinct from shuffling of input.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[BootstrapSample](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/BootstrapSample.cs)]
+        /// ]]>
+        /// </format>
+        /// </example>
+        public IDataView BootstrapSample(IDataView input,
+            bool complement = BootstrapSamplingTransformer.Defaults.Complement,
+            uint? seed = null,
```

Comment:
I think the seed is a more fundamental option than complement... it seems like whenever you *might* want to make use of a non-default argument of complement, you would need to use seed, whereas I can see situations where one might want to use seed and not *neecessarily* using complement. This suggests to me, seed goes first.

Reply:
Done.